Project-Hotpot/Project-Hotpot/Managers/SpotifyManager.swift

```diff
@@ -9,26 +9,28 @@ import Foundation
 import UIKit
 
 class SpotifyManager: NSObject {
-    
     //MARK: - Variables
     var currentSongLabel:String?
     var originalGenreSeeds: [String]
     var lastPlayerState: SPTAppRemotePlayerState?
     // MARK: - Spotify Authorization & Configuration
     var responseCode: String? {
         didSet {
-            fetchAccessToken { (dictionary, error) in
-                if let error = error {
+            fetchAccessToken { result in
+                switch result {
+                case .success(let dictionary):
+                    guard let accessToken = dictionary["access_token"] as? String else {
+                        return
+                    }
+                    DispatchQueue.main.async {
+                        self.appRemote.connectionParameters.accessToken = accessToken
+                        self.appRemote.connect()
+                    }
```

Comment:
Mmm should this be done on UI Thread? Does not look like any UI task

Reply:
So this is Spotify API Specific. If this part isn't run on the main thread, it often crashes. This is also why I put the play method on the main thread as well.

Reply:
<img width="1504" alt="Screen Shot 2022-08-01 at 4 46 35 PM" src="https://user-images.githubusercontent.com/107517438/182263627-3d6e7aaa-ad8f-4338-941d-2107d5095fb3.png">
Both the FetchContentItem and Play methods must be on the main thread.